### PR TITLE
Write some Tomes

### PR DESCRIPTION
**Pull request by petercombs**
Technically all the tomes.

### CODE REVIEW

File: eldritch/items/unique.py [hunk 1]
```diff
@@ -350,3 +350,97 @@ def get_interrupt(self, event, owner, state):
...
       return events.ChangeMovementPoints(owner, 3)
     return None
+
+# Tomes
+
+
+class TibetanTome(Tome):
+  def __init__(self, idx):
+    super().__init__("Tibetan Tome", idx, "unique", 3, 2)
+    self.max_tokens["stamina"] = 2
```
Comment (fingoldfish) [inline eldritch/items/unique.py:360]: Add a max tokens event?

File: eldritch/items/unique.py [hunk 2]
```diff
@@ -350,3 +350,97 @@ def get_interrupt(self, event, owner, state):
...
+
+  def read_event(self, owner):
+    check = events.Check(owner, "lore", -1)
+    success = events.Sequence([
+        events.Draw(owner, "spells", 1),
+        events.Loss(owner, {"sanity": 1}, source=self),
```
Comment (fingoldfish) [inline eldritch/items/unique.py:366]: I would recommend putting the token before the sanity loss - that way, if they go insane, they still put the token on the tome.

File: eldritch/items/unique.py [hunk 3]
```diff
@@ -350,3 +350,97 @@ def get_interrupt(self, event, owner, state):
...
+
+class BlackMagicTome(Tome):
+  def __init__(self, idx):
+    super().__init__("Black Magic Tome", idx, "unique", 3, 2)
+
+  def read_event(self, owner):
+    check = events.Check(owner, "lore", -2)
+    success = events.Sequence([
+        events.Draw(owner, "spells", 1),
+        events.GainOrLoss(owner, gains={"clues": 1}, losses={"sanity": 2}, source=self),
```
Comment (fingoldfish) [inline eldritch/items/unique.py:395]: Same here and below - discard before the loss.

File: eldritch/items/unique.py [hunk 4]
```diff
@@ -350,3 +350,97 @@ def get_interrupt(self, event, owner, state):
...
+        events.Draw(owner, "spells", 1),
+        events.Loss(owner, losses={"sanity": 1}, source=self),
+        events.DiscardSpecific(owner, [self])
+    ], owner
+    )
+    return events.PassFail(owner, check, success, events.Nothing())
+
+
+class BookOfTheDead(Tome):
+  "Necronomicon"
```
Comment (fingoldfish) [inline eldritch/items/unique.py:418]: Might have forgotten something here.

File: eldritch/items/unique.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
 import operator
 from eldritch import events, values
-from .base import Item, Weapon, OneshotWeapon
+from .base import Item, Weapon, OneshotWeapon, Tome
```
Comment (fingoldfish) [inline eldritch/items/unique.py:3]: Don't forget to add the new tomes to `__all__`, and also create them in the Create() functions. And also remove the TODO that references the tomes.

File: eldritch/items/unique.py [hunk 6]
```diff
@@ -350,3 +350,97 @@ def get_interrupt(self, event, owner, state):
...
+    super().__init__("Yellow Play", idx, "unique", 2, 2)
+
+  def read_event(self, owner):
+    check = events.Check(owner, "lore", -2)
+    success = events.Sequence([
+        events.GainOrLoss(owner, gains={"clues": 4}, losses={"sanity": 1}, source=self),
+        events.DiscardSpecific(owner, [self])
+    ], owner
+    )
+    return events.PassFail(owner, check, success, events.Nothing())
```
Comment (fingoldfish) [inline eldritch/items/unique.py:446]: Can I request that you add a test that covers all of the tomes? Something like
```
for tome in list_of_tomes:
  char.possessions = [tome]
  resolve_to_choice(CityChoice)
  assert tome.handle in state.usables
```
Would be good to get coverage on the creation of the read events.

**Review by fingoldfish — CHANGES_REQUESTED** [review]

File: eldritch/items/unique.py (hunk 1 above)
Comment (petercombs) [reply eldritch/items/unique.py:360]: `Item.get_max_token_event` already returns `DiscardSpecific(owner, [self])`

File: eldritch/items/common.py [hunk 7]
```diff
@@ -135,8 +135,20 @@ def read_event(self, owner):
     return events.PassFail(owner, check, success, events.Nothing())
 
 
+class OldJournal(Tome):
```
Comment (fingoldfish) [inline eldritch/items/common.py:138]: You forgot to add this to `__all__` and `CreateCommon`

**Review by fingoldfish — APPROVED** [review]

**Pull request merged**